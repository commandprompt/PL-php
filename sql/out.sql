CREATE SCHEMA plphp;

-- check an SRF with multiple OUT arguments.
CREATE OR REPLACE FUNCTION plphp.multiple_out(number OUT INTEGER, prefered OUT TEXT) RETURNS SETOF RECORD
AS $$
	$result = array(	
		array(0, 'zero'),
		array(1, 'ichi'),
		array(2, 'ni'),
		array(3, 'san'),
		array(4, 'yon'),
		array(5, 'go'),
		array(6, 'roku'),
		array(7, 'nana'),
		array(8, 'hachi'),
		array(9, 'ku'));
	for ($i = 0; $i < 9; $i++) {
		return_next($result[$i]);
	}
	return;
$$ LANGUAGE plphp;

SELECT * FROM plphp.multiple_out(); 

-- check that INOUT and OUT arguments can be used together.
CREATE OR REPLACE FUNCTION plphp.inout(s OUT INTEGER, a INOUT INTEGER, b INOUT INTEGER) 
AS $$
	$s = $a;
	$a = $b;
	$b = $s;
	$s += $a;
$$ LANGUAGE plphp;

SELECT * FROM plphp.inout(24, 42);

-- check TABLE arguments. Make sure that both return_next(array) and return_next()
-- work when TABLE arguments are declared.
CREATE OR REPLACE FUNCTION plphp.table_out(lim INTEGER) RETURNS TABLE(number INTEGER, square INTEGER) AS
$$
	for ($number = 1; $number <= $lim; $number++) {
		$square = $number * $number;
		if ($number % 2)
			return_next();
		else
			return_next(array($number, $square));
	}
$$ LANGUAGE plphp;

SELECT * FROM plphp.table_out(10);

-- check an OUT function returning invalid values 
CREATE OR REPLACE FUNCTION plphp.out_invalid(a OUT INTEGER, "long identifier" OUT INTEGER) AS
$$
 
  return(array(1,2));
$$ LANGUAGE plphp;

SELECT * FROM plphp.out_invalid();

