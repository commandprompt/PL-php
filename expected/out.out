CREATE SCHEMA plphp;
-- check an SRF with multiple OUT arguments.
CREATE OR REPLACE FUNCTION plphp.multiple_out(number OUT INTEGER, prefered OUT TEXT) RETURNS SETOF RECORD
AS $$
	$result = array(	
		array(0, 'zero'),
		array(1, 'ichi'),
		array(2, 'ni'),
		array(3, 'san'),
		array(4, 'yon'),
		array(5, 'go'),
		array(6, 'roku'),
		array(7, 'nana'),
		array(8, 'hachi'),
		array(9, 'ku'));
	for ($i = 0; $i < 9; $i++) {
		return_next($result[$i]);
	}
	return;
$$ LANGUAGE plphp;
SELECT * FROM plphp.multiple_out(); 
 number | prefered 
--------+----------
      0 | zero
      1 | ichi
      2 | ni
      3 | san
      4 | yon
      5 | go
      6 | roku
      7 | nana
      8 | hachi
(9 rows)

-- check that INOUT and OUT arguments can be used together.
CREATE OR REPLACE FUNCTION plphp.inout(s OUT INTEGER, a INOUT INTEGER, b INOUT INTEGER) 
AS $$
	$s = $a;
	$a = $b;
	$b = $s;
	$s += $a;
$$ LANGUAGE plphp;
SELECT * FROM plphp.inout(24, 42);
 s  | a  | b  
----+----+----
 66 | 42 | 24
(1 row)

-- check TABLE arguments. Make sure that both return_next(array) and return_next()
-- work when TABLE arguments are declared.
CREATE OR REPLACE FUNCTION plphp.table_out(lim INTEGER) RETURNS TABLE(number INTEGER, square INTEGER) AS
$$
	for ($number = 1; $number <= $lim; $number++) {
		$square = $number * $number;
		if ($number % 2)
			return_next();
		else
			return_next(array($number, $square));
	}
$$ LANGUAGE plphp;
SELECT * FROM plphp.table_out(10);
 number | square 
--------+--------
      1 |      1
      2 |      4
      3 |      9
      4 |     16
      5 |     25
      6 |     36
      7 |     49
      8 |     64
      9 |     81
     10 |    100
(10 rows)

-- check an OUT function returning invalid values 
CREATE OR REPLACE FUNCTION plphp.out_invalid(a OUT INTEGER, "long identifier" OUT INTEGER) AS
$$
 
  return(array(1,2));
$$ LANGUAGE plphp;
SELECT * FROM plphp.out_invalid();
ERROR:  "long identifier" can not be used as a PHP variable name
